pkg/syncer/k8scloudoperator/placement.go

```diff
@@ -776,3 +784,30 @@ func getHostCandidates(ctx context.Context, topologyRequirement *csi.TopologyReq
 	}
 	return hostNames
 }
+
+// getHostCandidates returns the hostnames where the placement can be done. Returns error if there is a misconfiguration
+func getHostCandidates(ctx context.Context, curPVC *v1.PersistentVolumeClaim, tops *csi.TopologyRequirement) ([]string, error) {
+	log := logger.GetLogger(ctx)
+	candidateHostsFromTopology := getHostNamesFromTopology(ctx, tops)
+
+	if affineToHostName, present := curPVC.ObjectMeta.Annotations[nodeAffinityAnnotationKey]; present {
+		log.Infof("Got affinity parameter host name: %s", affineToHostName)
+		// Abort placement if the affineToHostName is not present in the topology requirement
+		if !isHostPresentInTopology(affineToHostName, candidateHostsFromTopology) {
+			return nil, fmt.Errorf("invalid configuration - host specified in the nodeAffinity annotation is " +
+				"not provided in the topology requirement")
```

Comment:
Nit: The error message does not convey where to look for topology requirement. Could this read something like as:
```
invalid configuration - the PVC nodeAffinity annotation conflicts with Pod topology requirement
```